quinn-proto/src/connection/mod.rs

```diff
@@ -1162,9 +1162,13 @@ where
         if info.ack_eliciting && self.path.challenge.is_none() {
             // Only pass ACKs to the congestion controller if we are not validating the current
             // path, so as to ignore any ACKs from older paths still coming in.
-            self.path
-                .congestion
-                .on_ack(now, info.time_sent, info.size.into(), self.app_limited);
+            self.path.congestion.on_ack(
+                now,
+                info.time_sent,
+                info.size.into(),
+                self.app_limited,
+                self.path.rtt.get(),
```

Comment:
The parameter is called `min_rtt`, but this passes the smoothed RTT. Is this what is supposed to happen?

Reply:
From the equation on Wikipedia it seems like it should be neither. I got it from here https://github.com/cloudflare/quiche/blob/567cc5ebc530ed5cf2c14dd4af88a2496bacb290/src/recovery/cubic.rs#L286 which applies to the first equation here https://en.wikipedia.org/wiki/CUBIC_TCP. T is defined as "Time elapsed since the last window reduction". I'm not sure why min_rtt is used in quiche but I'm also not that familiar with CCAs. For what it's worth just observing the cwnd and congestion events behaves like I'd expect a cubic graph to look like from the examples so it can't be too far off either way.

Reply:
I think this is related to quiche's use of [HyStart](https://datatracker.ietf.org/doc/html/draft-ietf-tcpm-hystartplusplus). We should refer to that document for the proper parameters.

Reply:
When porting it over I assumed false for the hystart flag to simplify things and just focus on cubic, so this impl should not have hystart in it.

Reply:
> Time elapsed since the last window reduction

That definition of T doesn’t sound related to RTT at all to me? window is reduced when there is a congestion event, so is this the time since the last congestion event? 

Or does reduced also apply for „using“ parts of the congestion window by sending data. But I doubt it, with pacing that time is more or less constant.

Reply:
so I had a closer look at the equations and code, it looks like the min_rtt should be removed from the w_cubic function since T is already covered in that first arg by the literal t variable. I'm not sure where w_est came from though. the comments say it came from equation 4. still looking.

Reply:
found the 4th equation. rtt should be there, but not min rtt. https://datatracker.ietf.org/doc/html/rfc8312#section-4.2

Reply:
I'm not sure why `min_rtt` is added here, but I guess there was a reason. 
Note that there is an update to the Cubic spec which Quiche already seems to follow: 

https://github.com/NTAP/rfc8312bis/pull/3

https://github.com/NTAP/rfc8312bis/blob/main/draft-ietf-tcpm-rfc8312bis.md

Maybe this one is different from Wikipedia?

Reply:
https://github.com/NTAP/rfc8312bis/pull/3/files#diff-b956a53223cc21d8f0bc2affbb81d6f32e0eb24f3a38f494a53af70e9dd070acR393 in line 393 adds the rtt, but doesn't specify which one. 

And for RTT the spec describes:

> RTT: Smoothed round-trip time in seconds, calculated as described in {{!RFC6298}}.

So it seems like passing the smoothed RTT is correct (even though I don't know whether the smoothing is the same between QUIC reocvery and that spec).

In that case I would remember just renaming the parameter from `min_rtt` to to just `rtt` or `smoothed_rtt`

Reply:
here's a rendered version of the new spec: https://ntap.github.io/rfc8312bis/draft-ietf-tcpm-rfc8312bis.html#name-introduction

Reply:
👍  rtt it is

Reply:
@Matthias247 good to close this thread?

Reply:
If the implementation follows the new draft - sure